src/primitive_file_win.cc

```diff
@@ -92,65 +92,68 @@ static const int FILE_ST_ATIME = 8;
+  word length = GetCurrentDirectoryW(0, NULL);
   if (length == 0) {
     FATAL("Failed to get current dir");
   }
-  current_directory = reinterpret_cast<char*>(malloc(length));
-  if (!current_directory) return null;
-  if (GetCurrentDirectory(length, const_cast<char*>(current_directory)) == 0) {
+  current_directory = unvoid_cast<wchar_t*>(malloc(length * sizeof(wchar_t)));
+  if (GetCurrentDirectoryW(length, const_cast<wchar_t*>(current_directory)) == 0) {
     FATAL("Failed to get current dir");
   }
   process->set_current_directory(current_directory);
   return current_directory;
 }
 
-HeapObject* get_absolute_path(Process* process, const char* pathname, char* output) {
-  size_t pathname_length = strlen(pathname);
+HeapObject* get_absolute_path(Process* process, const wchar_t* pathname, wchar_t* output) {
+  size_t pathname_length = wcslen(pathname);
 
-  // Poor man's version. For better platform handling, use UNICODE and PathCchAppendEx.
+  // Poor man's version. For better platform handling, use PathCchAppendEx.
   // TODO(florian): we should probably use PathCchCombine here. That would remove
   // all the special checks.
 
-  if (!PathIsRelative(pathname)) {
-    if (GetFullPathName(pathname, MAX_PATH, output, NULL) == 0) WINDOWS_ERROR;
+  if (!PathIsRelativeW(pathname)) {
+    if (GetFullPathNameW(pathname, MAX_PATH, output, NULL) == 0) WINDOWS_ERROR;
     return null;
   }
 
-  const char* current_directory = current_dir(process);
-  if (!current_directory) MALLOC_FAILED;
+  const wchar_t* current_directory = current_dir(process);
 
   // Check if the path is rooted. On Windows paths might not be absolute, but
   // relative to the drive/root of the current working directory.
   // For example the path `\foo\bar` is a rooted path which is relative to
   // the drive of the current working directory.
-  char root[MAX_PATH];
-  const char* relative_to = null;
+  wchar_t root[MAX_PATH];
+  const wchar_t* relative_to = null;
   if (pathname_length > 0 && (pathname[0] == '\\' || pathname[0] == '/')) {
     // Relative to the root of the drive/share.
     // For example '\foo\bar' is rooted to the current directory's drive.
-    strncpy(root, current_directory, MAX_PATH);
+    wcsncpy(root, current_directory, MAX_PATH);
     root[MAX_PATH - 1] = '\0';
-    if (!PathStripToRoot(root)) WINDOWS_ERROR;
+    if (!PathStripToRootW(root)) WINDOWS_ERROR;
     relative_to = root;
   } else {
     relative_to = current_directory;
   }
 
-  char temp[MAX_PATH];
-  if (snprintf(temp, MAX_PATH, "%s\\%s", relative_to, pathname) >= MAX_PATH) INVALID_ARGUMENT;
-  if (GetFullPathName(temp, MAX_PATH, output, NULL) == 0) WINDOWS_ERROR;
+  wchar_t temp[MAX_PATH];
+  if (snwprintf(temp, MAX_PATH, L"%ls\\%ls", relative_to, pathname) >= MAX_PATH) INVALID_ARGUMENT;
+  if (GetFullPathNameW(temp, MAX_PATH, output, NULL) == 0) WINDOWS_ERROR;
   return null;
 }
 
+#define BLOB_TO_ABSOLUTE_PATH(result, blob)                        \
```

Comment:
Add a comment, that almost every function should use this macro, as we use a per-process working directory. As such we can never give a relative path to the system function.

On Posix, we change the paths already in Toit land (I guess to avoid allocations). Maybe/probably we should add checks in all primitives that the incoming paths are indeed absolute.